feat(username): implement username availability check with input component

abhishek-nexgen-dev opened the pull request:
Issue Number : - #37

abhishek-nexgen-dev committed:
feat(auth): add NavBar component to login and signup pages

abhishek-nexgen-dev committed:
feat(username): implement username availability check with input component

sanjaysah101 committed:
Merge branch '33-upgrade-packages' of github.com:ansopedia/creator-studio into pr/abhishek-nexgen-dev/40

sanjaysah101 commented on src/app/(auth)/login/page.tsx:
```diff
@@ -1,35 +1,39 @@
+        </div>
       </div>
-    </div>
+    </>
```
Revert these file changes. We don't need to addthe  Navbar explicitly in the login page. It is already in (auth)/layout.tsx

sanjaysah101 commented on src/app/(auth)/login/page.tsx:
```diff
@@ -1,35 +1,39 @@
+            Welcome Back
+            <Typography variant="span" className="text-primary">
+              &nbsp;Ansopedian!
+            </Typography>
           </Typography>
-        </Typography>
-        <Typography className="mt-2">Login to continue</Typography>
-        <LoginForm />
-        <Divider>Or</Divider>
-        <SignInWithGoogle />
-        <Typography className="mt-2">
-          Don&apos;t have an account yet?
-          <Link href="/signup" className="text-blue-500 hover:underline">
-            &nbsp;Sign up
-          </Link>
-        </Typography>
-      </div>
-      <div className="hidden items-center justify-center sm:block sm:w-2/3 md:w-2/3 lg:w-1/3">
-        <Image src="/images/login-illustrator.svg" alt="login" width={500} height={500} priority />
+          <Typography className="mt-2">Login to continue</Typography>
+          <LoginForm />
+          <Divider>Or</Divider>
+          <SignInWithGoogle />
+          <Typography className="mt-2">
+            Don&apos;t have an account yet?
+            <Link href="/signup" className="text-blue-500 hover:underline">
+              &nbsp;Sign up
+            </Link>
+          </Typography>
+        </div>
+        <div className="hidden items-center justify-center sm:block sm:w-2/3 md:w-2/3 lg:w-1/3">
+          <Image src="/images/login-illustrator.svg" alt="login" width={500} height={500} priority />
+        </div>
       </div>
-    </div>
+    </>
   );
 };
 
```
Revert this file. Since we have already Navbar in its layout.

sanjaysah101 commented on src/constants/form.constants.ts:
```diff
@@ -12,13 +12,13 @@ export interface FormFieldConfig extends FieldValues {
-    name: "username",
-    type: "text",
-  },
+  // {
+  //   id: "signup_username",
+  //   inputType: "input",
+  //   placeholder: "Create an unique username",
+  //   name: "username",
+  //   type: "text",
+  // },
```
By commenting this out. The username is not displayed on the signup page.

sanjaysah101 commented on src/lib/services/authService.ts:
```diff
@@ -1,9 +1,11 @@
 "use server";
 
+import { console } from "inspector";
```
Remove inspector library. We don't need it. native console will work fine

abhishek-nexgen-dev replied:
Okay I will fix it tomorrow

sanjaysah101 commented on src/lib/services/authService.ts:
```diff
@@ -22,6 +24,14 @@ export const signup = async (body: SignUpSchema): Promise<IApiResponse<SignUpRes
   return POST(url, { body });
 };
 
+export const checkUsernameAvailability = async (data: {
+  username: string;
+}): Promise<IApiResponse<{ available: boolean }>> => {
```
It should be data: {isAvailable: boolean}

abhishek-nexgen-dev replied:
Okay

abhishek-nexgen-dev committed:
feat(auth): remove NavBar from login and signup pages, add it to AuthProvider, and update username availability check response structure

abhishek-nexgen-dev committed:
feat: Implement courses management dashboard with CRUD functionality

- Add CoursesTable component for displaying courses with edit and delete actions.
- Create CreateCourseDialog component for adding new courses.
- Develop CoursesPage to manage course listings and integrate CoursesDashboard.
- Establish layout for the dashboard with Sidebar and Navbar components.
- Introduce Profile and Settings pages with forms for user profile management.
- Implement Logout functionality to clear user session.
- Enhance user experience with loading states and error handling in profile and user pages.
- Add CopyrightFooter component for consistent footer across the application.

abhishek-nexgen-dev committed:
feat(layout): replace AuthNavBar with NavBar in PublicLayout and update import paths

abhishek-nexgen-dev committed:
feat(ui): enhance UsernameInput component with props for controlled behavior and availability check callback

abhishek-nexgen-dev commented:
@sanjaysah101

sanjaysah101 commented on src/app/(auth)/layout.tsx:
```diff
@@ -1,9 +1,9 @@
-import AuthNavBar from "./components/AuthNavBar";
+import NavBar from "@/components/NavBar";
 
 export default function PublicLayout({ children }: { children: React.ReactNode }) {
   return (
     <>
-      <AuthNavBar />
+      <NavBar />
```
It should be `AuthNavBar`

abhishek-nexgen-dev replied:
But `Navbar` is Batter than Auth NavBar

sanjaysah101 replied:
In the Navbar component there is `searchbar` that we don't need here. Also, there is an Auth user indicator (Profile) and CTA buttons that also, we don't need in auth pages. Only Non-Authenticated users should access these pages.

abhishek-nexgen-dev replied:
Okay I will fix it

sanjaysah101 commented on src/app/(auth)/login/page.tsx:
```diff
@@ -7,29 +7,31 @@ import { LoginForm } from "../components";
+        </div>
       </div>
-    </div>
+    </>
```
Why have you used React.Fragment. It is not need here

sanjaysah101 commented on src/app/(protected)/layout.tsx:
```diff
@@ -1,12 +1,7 @@
   return (
     <>
-      <NavBar />
       <main className="min-h-[calc(100vh-64px)]">{children}</main>
-      <Footer />
     </>
```
Remove this fragment

abhishek-nexgen-dev committed:
feat(layout): replace NavBar with AuthNavBar in PublicLayout and clean up unnecessary fragments in layout components

abhishek-nexgen-dev commented:
@sanjaysah101
